김현국/app/src/main/java/com/example/myapplication/adapter/RepoAdapter.kt

```diff
@@ -0,0 +1,79 @@
+package com.example.myapplication.adapter
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.example.myapplication.databinding.ItemSearchBinding
+import com.example.myapplication.model.Repository
+import com.example.myapplication.presenter.search.OnItemClick
+import com.example.myapplication.presenter.search.SearchAdapterContract
+
+/**
+ * @author 김현국
+ * @created 2022/04/25
+ */
+class RepoAdapter() :
+    ListAdapter<Repository, RepoAdapter.RepoViewHolder>(diffUtil),
+    SearchAdapterContract.View,
+    SearchAdapterContract.Model {
+
+    lateinit var onItemClick: OnItemClick
```

Comment:
itemClick 리스너를 무조건 등록한다는 조건이라면 `lateinit var`보다는 생성자에서 주입받게 해도 될 것 같아요!
```suggestion
class RepoAdapter(
     private val onItemClick: OnItemClick
    ) :
    ListAdapter<Repository, RepoAdapter.RepoViewHolder>(diffUtil),
    SearchAdapterContract.View,
    SearchAdapterContract.Model {
```

Reply:
이거는 마땅한 방법을 못찾아서 mvvm 구조에서 변경해볼께용